src/hotspot/cpu/x86/x86.ad

```diff
@@ -6089,6 +6095,36 @@ instruct signumD_reg(regD dst, regD zero, regD one, rRegP scratch, rFlagsReg cr)
   ins_pipe( pipe_slow );
 %}
 
+instruct signumV_reg_avx(vec dst, vec src, vec zero, vec one, vec xtmp1) %{
+  predicate(Matcher::vector_length_in_bytes(n) < 64);
+  match(Set dst (SignumVF src (Binary zero one)));
+  match(Set dst (SignumVD src (Binary zero one)));
+  effect(TEMP dst, TEMP xtmp1);
+  format %{ "vector_signum_avx $dst, $src\t! using $zero, $one and $xtmp1 as TEMP" %}
+  ins_encode %{
+    int opcode = this->ideal_Opcode();
+    int vec_enc = vector_length_encoding(this);
+    __ vector_signum_avx(opcode, $dst$$XMMRegister, $src$$XMMRegister, $zero$$XMMRegister, $one$$XMMRegister,
+                         $xtmp1$$XMMRegister, vec_enc);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct signumV_reg_evex(vec dst, vec src, vec zero, vec one, kReg ktmp1) %{
+  predicate(Matcher::vector_length_in_bytes(n) ==  64);
```

Comment:
avx512vl check is needed here. vector_signum_evex needs avx512vl support.

Reply:
Further clarification here: The vector_signum_evex code is faster than vector_signum_vex. On AVX=3 platforms vector_signum_evex should be used for all vector lengths.
So the predicate for signumV_reg_evex should be:
    ((VM_Version::supports_avx512vl()) || (Matcher::vector_length_in_bytes(n) == 64))
Accordingly the predicate for signumV_reg_avx should be adjusted to be reverse of this.

Reply:
I removed it because instruction sequence for AVX2 saves an extra vector compare instruction,  so we can emit AVX512 sequence only in case of 64 byte vector operation.